### ISSUE TITLE
Template application: `pgvector`

### ISSUE THREAD

**Issue by edmondwinston** [railwayapp/templates#460]
### Contact Details

edmondwinston.personal@gmail.com

### Template URL

https://railway.app/template/uYYYQy

### What does this template deploy?

This template deploys a Postgres instance with `pgvector` extension enabled.

### Demo project

https://railway.app/project/ff99ab66-c60f-4301-b787-ab9c0b7c3969

**Comment by imsumoregn** [comment]
The @see in the demo project, great idea, but also put it in an `example` directory, for easy future references.

**Comment by brody192** [comment]
Little tip, instead of having the user define their own password, use the `secret()` [template variable function](https://docs.railway.app/reference/templates#template-variable-functions)

**Comment by edmondwinston** [comment]
@imsumoregn
> The @see in the demo project, great idea, but also put it in an `example` directory, for easy future references.

Yb, but could you also walk me through steps to setup `nginx` in our next US? I know it should be simple, but I'm new to the game and it's not a pile of water, it's a whole sea I'm looking at.

@brody192
> Little tip, instead of having the user define their own password, use the `secret()` [template variable function](https://docs.railway.app/reference/templates#template-variable-functions)

Ah, I should've read the docs more carefully. Thank you for the tip.

**Comment by imsumoregn** [comment]
Sure, put it in the agenda, please. Nginx is quite something - it is dead simple on its own, but the hard part is how, where, and why you would use it.

**Comment by imsumoregn** [comment]
But the next session is not until a while, do you feel like implement this reverse proxy thing with Caddy for the time being? The other mate in this thread (Hey Brody! You are everywhere) has submitted a few templates that also used Caddy, if the workshop last month didn't help you.

**Comment by brody192** [comment]
I'm sure this goes against what the team thinks, but they shouldn't have asked for database templates until Railway supports TCP proxying, because unless you use some type of tunnel with this template, it's unusable from anywhere other than the private network

**Comment by edmondwinston** [comment]
@brody192 It really is! I don't understand a word from what you've said, but I've been trying to expose the db using your reverse proxy Caddyfile, and it didn't work.
cc: @imsumoregn.

**Comment by imsumoregn** [comment]
Yes, ngrok or the good old localtunnel would work. I'm with Brody on the unnecessity of implementing this, but if you just want to have fun, it shouldn't be a problem.

Also, don't turn all your notification off, please. At least not Slack.

**Comment by edmondwinston** [comment]
I saw your message, really sorry. I'm already trying `ngrok` and will open another issue for the true standalone version when finish.

**Comment by edmondwinston** [comment]
Closing because Railway has released TCP proxy and already built a `pgvector` template as an example. Great news, btw.
